addons/mail/static/src/core/common/thread_service.js

```diff
@@ -851,45 +852,60 @@ export class ThreadService {
     }
 
     /**
-     * @param {import("models").Persona} persona
-     * @param {import("models").Thread} [thread]
+     * @param {{ persona: import("@mail/core/persona_model").Persona, thread: import("@mail/core/thread_model").Thread }}
      */
-    avatarUrl(persona, thread) {
-        if (!persona) {
-            return DEFAULT_AVATAR;
-        }
-        const urlParams = {};
-        if (persona.write_date) {
-            urlParams.unique = persona.write_date;
-        }
-        if (persona.is_company === undefined && this.store.self?.user?.isInternalUser) {
-            this.personaService.fetchIsCompany(persona);
-        }
-        if (thread?.model === "discuss.channel") {
-            if (persona.type === "partner") {
-                return url(`/discuss/channel/${thread.id}/partner/${persona.id}/avatar_128`, urlParams);
+    avatarUrl({ persona, thread }) {
```

Comment:
Naming the params is making the method harder to use.
Thread should always be required, and it should be the first param.